emperor/support_files/js/color-view-controller.js

```diff
@@ -56,62 +55,108 @@ define([
-        // we re-use this same callback regardless of whether the
-        // color or the metadata category changed, maybe we can do
-        // something better about this
-        var category = scope.$select.val();
-        var colorScheme = scope.$colormapSelect.val();
-
-        var k = scope.getActiveDecompViewKey();
-        var decompViewDict = scope.decompViewDict[k];
-
-        // getting all unique values per categories
-        var uniqueVals = decompViewDict.decomp.getUniqueValuesByCategory(category);
-        // getting color for each uniqueVals
-        var attributes = ColorViewController.getColorList(uniqueVals, colorScheme);
-        // fetch the slickgrid-formatted data
-        var data = decompViewDict.setCategory(attributes, ColorViewController.setPlottableAttributes, category);
-
-        scope.setSlickGridDataset(data);
-      },
-      'slickGridColumn':{id: 'title', name: '', field: 'value',
+    var options = {
+      'valueUpdatedCallback':
+        function(e, args) {
+          var val = args.item.category, color = args.item.value;
+          var group = args.item.plottables;
+          var element = scope.decompViewDict[scope.getActiveDecompViewKey()];
+          ColorViewController.setPlottableAttributes(element, color, group);
+        },
+      'categorySelectionCallback':
+        function(evt, params) {
+          // we re-use this same callback regardless of whether the
+          // color or the metadata category changed, maybe we can do
+          // something better about this
+          var category = scope.$select.val();
+          var discrete = $('option:selected', scope.$colormapSelect)
+                           .attr('data-type') == 'Discrete';
+          var colorScheme = scope.$colormapSelect.val();
+
+          var k = scope.getActiveDecompViewKey();
+          var decompViewDict = scope.decompViewDict[k];
+
+          if (discrete) {
+            scope.$scaled.prop('checked', false);
+            scope.$scaled.prop('disabled', true);
+          } else {
+            scope.$scaled.prop('disabled', false);
+          }
+          var scaled = scope.$scaled.is(':checked');
+          // getting all unique values per categories
+          var uniqueVals = decompViewDict.decomp.getUniqueValuesByCategory(category);
+          // getting color for each uniqueVals
+          var colorInfo = ColorViewController.getColorList(uniqueVals, colorScheme, discrete, scaled);
+          var attributes = colorInfo[0];
+          // fetch the slickgrid-formatted data
+          var data = decompViewDict.setCategory(attributes, ColorViewController.setPlottableAttributes, category);
+
+          if (scaled) {
+            scope.setSlickGridDataset({});
+            scope.$gridDiv.hide();
+            scope.$scaleDiv.show();
+            scope.$colorScale.show();
+            scope.$colorScale.html(colorInfo[1]);
+          } else {
+            scope.setSlickGridDataset(data);
+            scope.$scaleDiv.hide();
```

Comment:
They are reversed, so this is hiding the scale and showing the slickgrid, while the if above does the opposite.